cin_validator/rules/cin2022_23/rule_8736.py

```diff
@@ -0,0 +1,188 @@
+    ref_date_series = header[ReferenceDate]
+    collection_start, collection_end = make_census_period(ref_date_series)
+
+    # Before you begin, rename the index so that the initial row positions can be kept intact.
+    df.index.name = "ROW_ID"
+
+    # lOGIC
+    # Implement rule logic as described by the Github issue.
+    # Put the description as a comment above the implementation as shown.
+
+    # Where present, if an <Assessments> group does not contain the <AssessmentAuthorisationDate> (N00160) then the <AssessmentActualStartDate> (N00159) must be on or between [Start_Of_Census_Year] and <ReferenceDate> (N00603)
+    condition_1 = df[AssessmentAuthorisationDate].isna
+    condition_2 = (df[AssessmentActualStartDate] >= collection_start) & (
+        df[AssessmentActualStartDate] <= collection_end
+    )
+
+    # get all the data that fits the failing condition. Reset the index so that ROW_ID now becomes a column of df
+    df_issues = df[condition_1 & condition_2].reset_index()
+
+    # SUBMIT ERRORS
+    # Generate a unique ID for each instance of an error. In this case,
+    # - If only LAchildID is used as an identifier, multiple instances of the error on a child will be understood as 1 instance.
+    # We don't want that because in reality, a child can have multiple instances of an error.
+    # - If we use the LAchildID-CPPstartDate combination, that artificially cancels out the instances where a start date repeats for the same child.
+    # Another rule checks for that condition. Not this one.
+    # - It is very unlikely that a combination of LAchildID-CPPstartDate-CPPendDate will repeat in the DataFrame.
+    # Hence, it can be used as a unique identifier of the row.
+
+    # Replace CPPstartDate and CPPendDate below with the columns concerned in your rule.
+    link_id = tuple(
+        zip(
+            df_issues[AssessmentAuthorisationDate], df_issues[AssessmentActualStartDate]
+        )
+    )
+    df_issues["ERROR_ID"] = link_id
+    df_issues = df_issues.groupby("ERROR_ID")["ROW_ID"].apply(list).reset_index()
+    # Ensure that you do not change the ROW_ID, and ERROR_ID column names which are shown above. They are keywords in this project.
+    rule_context.push_type_1(
+        table=Assessments,
+        columns=[AssessmentAuthorisationDate, AssessmentActualStartDate],
+        row_df=df_issues,
+    )
+
+
+def test_validate():
+    # Create some sample data such that some values pass the validation and some fail.
+
+    fake_header = pd.DataFrame(
+        [{ReferenceDate: "31/03/2022"}]  # the census start date here will be 01/04/2021
+    )
+
+    child_assessments = pd.DataFrame(
+        [
+            {
+                "LAchildID": "child1",
+                "AssessmentAuthorisationDate": pd.NA,
+                "AssessmentActualStartDate": "27/05/2021",
+            },
+            {
+                "LAchildID": "child2",
+                "AssessmentAuthorisationDate": "24/02/2021",
```

Comment:
You'll need to add CINdetailsIDs to these now, because of the new condition_1.

Also, the rule checks if StartDAtes are out of bound where there is no Authorisation, so the failing rows will need to have pd.NAs.